cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -122,6 +125,19 @@ public void start() {
       Threads.createDaemonThreadFactory("polling-booking-executor"));
 
     store = storeFactory.create();
+
+    scheduleStore.initialize();
+
+    // Reschedule the persisted schedules
+    synchronized (this) {
```

Comment:
Yes, that was the point, although now that you mention it I think this can't happen, since we have to wait till the scheduler services starts up before doing any operation.